.github/workflows/tests.yml

```diff
@@ -0,0 +1,130 @@
+        coverage report
+    - name: Create/lint default config
+      env:
+        AWS_ACCOUNT_ID: ${{ secrets.INFRA_ACCOUNT_ID }}
+        GITHUB_SHA: ${{ github.sha }}
+        REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
+        REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+        GCR_JSON_CREDS: ${{ secrets.GCR_JSON_CREDS }}
+        ACM_CERT_ARN: ${{ secrets.ACM_CERT_ARN }}
+        BASE_DOMAIN: ${{ secrets.BASE_DOMAIN }}
+      run: |
+        export NAME=cdk-deploy-$(echo $GITHUB_SHA | cut -c1-6)
+        ./util.py generate_config_template --name $NAME --aws-region=us-west-2 --aws-account-id=$AWS_ACCOUNT_ID --dev --registry-username $REGISTRY_USERNAME --registry-password $REGISTRY_PASSWORD --gcr-json-creds $GCR_JSON_CREDS --hostname $NAME.$BASE_DOMAIN --acm-cert-arn $ACM_CERT_ARN --disable-flow-logs > config.yaml
+        ./util.py load_config -f ./config.yaml
+    - name: Test default config (single and nested stacks)
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.INFRA_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.INFRA_ACCESS_KEY }}
+      run: |
+        cdk synth --context singlestack=true -q
+        cdk synth -q
+    - name: Upload distribution artifacts
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.DOMINO_ARTIFACTS_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.DOMINO_ARTIFACTS_ACCESS_KEY }}
+        DOMINO_CDK_VERSION: ${{ github.sha }}
+        DATEDIR: "date +%Y%m%d"
+      run: |
+        cd ..
+        make clean && make dist
+        for suffix in "" "-terraform"; do
+        filename="domino-cdk$suffix-$DOMINO_CDK_VERSION.tar.gz"
+        aws s3 cp --acl=public-read ./dist/$filename s3://domino-artifacts/cdk/$($DATEDIR)/$filename
+        echo "Artifact url: https://domino-artifacts.s3.amazonaws.com/cdk/$($DATEDIR)/$filename"
+        done
+    - name: Deploy CDK
+      if: contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master'
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.INFRA_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.INFRA_ACCESS_KEY }}
+        REGISTRY_USERNAME: ${{ secrets.REGISTRY_USERNAME }}
+        REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+      run: |
+        docker login -u $REGISTRY_USERNAME -p $REGISTRY_PASSWORD quay.io
+        cdk deploy --require-approval never --outputs-file outputs.json
+    - name: Prepare Domino Install
+      if: contains(github.event.pull_request.labels.*.name, 'deploy-test') || github.ref == 'refs/heads/master'
+      env:
+        AWS_ACCESS_KEY_ID: ${{ secrets.INFRA_KEY_ID }}
+        AWS_SECRET_ACCESS_KEY: ${{ secrets.INFRA_ACCESS_KEY }}
+      run: |
+        $(jq -r ".[].ekskubeconfigcmd" outputs.json) --kubeconfig ./kubeconfig
+        jq -r ".[].agentconfig" outputs.json > agent_template.yaml
+        docker run --rm -v $(pwd):/cdk quay.io/domino/deployer:v42 python -m fleetcommand_agent init --full -t /cdk/agent_template.yaml -f /cdk/domino.yml
```

Comment:
Why deployer vs. fc-agent? Should we extract the tag somewhere or document when/if to update it?

Reply:
The agent doesn't have awscli in it, so the agent image can't be used to do the install phase since the kubeconfig won't be able to auth. Since the agent isn't ever used to install eks from an external source in practice (without the deployer) and at least needs some minor modification (refining the kubeconfig-searching code and come up with some new operational practices), it didn't seem worth adding awscli to the image right this second when there was an alternative available.

Also had some notions of adding a force destroy afterward later, and that may be in the deployer image already--though with moving it into an account that gets wiped regularly I don't really care so much about that anymore.